doc/design/refactor/distributed_architecture.md

```diff
@@ -110,30 +112,88 @@ img, label = input[0], input[1]
 hidden = paddle.layer.fc(input=img, size=200, act=paddle.activation.Tanh())
 prediction = paddle.layer.fc(input=img, size=10, act=paddle.activation.Softmax())
 cost = paddle.layer.classification_cost(input=prediction, label=label)
-optimizer = paddle.optimizer.SGD(cost, learning_rate=0.01)
-session = paddle.session.NewRemote(num_trainer=3, num_ps=2, GPU_per_trainer=1)
+optimizer = paddle.optimizer.SGD(learning_rate=0.01)
+opts = optimizer.minimize(cost)
+exe = RemoteExecutor(num_trainer=3, num_ps=2, GPU_per_trainer=2, sync_batches=1)
+# this will init variable data on both server and trainer
+exe.run(framework.default_startup_program())
+exe.sync()
+
 for i in range(1000):
-	_, cost_val = session.eval(targets=[cost, optimizer])
-	print cost_val
+  # feed data
+  ...
+  cost, acc = exe.run(framework.default_main_program(),
+                      fetch_list=[avg_cost, acc_out])
+  print cost, acc
 ```
 
 The code above is a typical Python trainer code, the neural network
 topology is built using helper functions such as
-`paddle.layer.fc`. The training is done by calling `session.eval`
+`paddle.layer.fc`. The training is done by calling `Executor.run`
 iteratively.
 
-#### session.eval
-
-As shown in the graph, `session.eval` sends the IR and the evaluation
-inputs/targets to the PaddlePaddle cluster for evaluation. The
-targets can be any variable in the computation graph. When the target
-is the `optimizer` variable, the neural network will be optimized
-once. When the target is the `cost` variable, `session.eval` returns
-the cost value.
+#### RemoteExecutor
```

Comment:
Where is the RemoteExecutor in? The master node of a cluster?
And where is the Converter in? Local machine?

Reply:
@QiJune @helinwang @putcn 

After a short discussion with @putcn  today, we may need more discussion to decide the API. Things we agree is that:

1. local/remote training should be of the same form.
1. we should  have a Python API to submit `ProgramDesc` to the cluster

The implementation may have two major methods:

1. start a server process (orchestrator/controller) in the cluster somewhere, it accepts `ProgramDesc` and execute it.
1.  submit a Python program as a cluster job. The Python program executes the `ProgramDesc`.